feat: add `complex/float64/base/assert/is-almost-equal`

anandkaranubc opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the implementation for `complex/float64/base/assert/is-almost-equal`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolve none.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add `complex/float64/base/assert/is-almost-equal-value`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/docs/repl.txt:
```diff
@@ -0,0 +1,41 @@
+    Examples
+    --------
+    > var z1 = new {{alias:@stdlib/complex/float64/ctor}}( 1.0, 3.0 );
+    > var z2 = new {{alias:@stdlib/complex/float64/ctor}}( 1.0, 3.0 );
```
Any suggestions here for including more examples with `EPS` without exceeding the word limit?

Can I declare `EPS` as:

`var EPS = {{alias:@stdlib/constants/float64/eps}}`

and then use it below?

anandkaranubc replied:
@kgryte

kgryte replied:
Updated.

kgryte commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/README.md:
````diff
@@ -0,0 +1,158 @@
+// returns true
+```
+
+The function returns `false` if either input value has a `NaN` real or imaginary part.
````
```suggestion
The function returns `false` if either input value has a `NaN` real or imaginary component.
```

kgryte committed:
docs: use conventional verbage

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/docs/repl.txt:
```diff
@@ -0,0 +1,41 @@
+    place).
+
+    The function returns `false` if either input value has a `NaN` real or
+    imaginary part.
```
```suggestion
    imaginary component.
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/docs/types/index.d.ts:
```diff
@@ -0,0 +1,54 @@
+*
+* ## Notes
+*
+* -   The function returns `false` if either input value has a `NaN` real or imaginary part.
```
```suggestion
* -   The function returns `false` if either input value has a `NaN` real or imaginary component.
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/lib/main.js:
```diff
@@ -0,0 +1,67 @@
+*
+* ## Notes
+*
+* -   The function returns `false` if either input value has a `NaN` real or imaginary part.
```
```suggestion
* -   The function returns `false` if either input value has a `NaN` real or imaginary component.
```

kgryte committed:
docs: update note

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/test/test.js:
```diff
@@ -0,0 +1,125 @@
+	t.end();
+});
+
+tape( 'the function returns `false` if provided `NaN` as either real or imaginary part', function test( t ) {
```
```suggestion
tape( 'the function returns `false` if provided `NaN` as either real or imaginary component', function test( t ) {
```

kgryte committed:
test: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/complex/float64/base/assert/is-almost-equal-value/test/test.js:
```diff
@@ -0,0 +1,125 @@
+	t.end();
+});
+
+tape( 'the function returns `true` if signed zeros are provided as real or imaginary parts irrespective of the specified number of ULPs', function test( t ) {
```
```suggestion
tape( 'the function returns `true` if signed zeros are provided as real or imaginary components irrespective of the specified number of ULPs', function test( t ) {
```

kgryte committed:
test: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: update examples

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (dismissed):
LGTM after updating the terminology to be consistent with elsewhere in the project.

anandkaranubc committed:
fix: rename function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Merge branch 'develop' into feat/is-almost-equal-c128

anandkaranubc committed:
fix: rename function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---
